src/Engine/Utils.java

```diff
@@ -531,11 +511,13 @@ public int getCachedDistance(XYCoord xyc)
         end = fixedEndpoint;
       }
 
-      final boolean theoretical = true;
-      final GamePath path = findShortestPath(start, myUnit, end, myMap, theoretical);
-      int distance = path.getFuelCost(myUnit, myMap);
-      if( 0 == path.getPathLength() )
-        distance = Integer.MAX_VALUE;
+      Utils.PathCalcParams pcp = new Utils.PathCalcParams(myUnit, myMap);
+      pcp.setTheoretical();
+      pcp.start = start;
+      final GamePath path = pcp.findShortestPath(end);
+      int distance = Integer.MAX_VALUE;
+      if( null != path )
+        distance = path.getFuelCost(myUnit, myMap);
```

Comment:
Did our zero-length path behavior change to return a null `Path` instead of an empty one?

Reply:
Okay, based on the next commit it looks like it.

Reply:
Yeah, I was tired of having two failure modes.